charts/hawkbit/templates/deployment.yaml

```
@@ -37,7 +37,11 @@ spec:
             - name: SPRING_PROFILES_ACTIVE
               value: "{{ .Values.spring.profiles }}"
             - name: "SPRING_DATASOURCE_URL"
+              {{- if .Values.env.springDatasourceUrl }}
+              value: "{{ .Values.env.springDatasourceUrl }}"
+              {{ else }}
```

```suggestion
              {{- else }}
```
if you have set env.springDatasourceUrl then in the result deployment file you will have an empty line after "value: ...". 
better to change it to "{{- else }}" in order to remove the excessive empty line